app/internal/timer.py

```diff
@@ -0,0 +1,34 @@
+from datetime import datetime
+from typing import Dict, List, Optional
+
+from app.routers.event import sort_by_date
+from app.routers.user import get_all_user_events
+from sqlalchemy.orm import Session
+from app.database.models import Event
+
+
+def get_next_user_event(session: Session, user_id: int) -> Optional[Event]:
+    events = get_only_future_events(
+        sort_by_date(get_all_user_events(session, user_id))
+    )
+    if events:
+        return events[0]
```

Comment:
This can use lots of memory, and might be inefficient for large scale systems.
Try to use SQLAlchemy to retrieve the next user's event. It shouldn't be hard.

Reply:
Done